hermes/bin/src/runtime_extensions/hermes/sqlite/state.rs

```diff
@@ -0,0 +1,108 @@
+// cspell: words mapref
+
+//! Internal state implementation for the `SQLite` module.
+
+use std::collections::HashMap;
+
+use dashmap::{mapref::one::RefMut, DashMap};
+use once_cell::sync::Lazy;
+
+use crate::app::HermesAppName;
+
+/// Represents an individual state for a particular object.
+#[derive(Debug)]
+pub(crate) struct ResourceObjectState<T: ToOwned + PartialEq> {
```

Comment:
I think it's not necessary to make a generic this struct right now.
I think it would be enough just to provide a type alias for `usize`, something like this
```Rust
type Pointer = usize;
```